keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -651,13 +610,119 @@ endpoints:
+  `ClusterIP` service, but the way that they are exposed depends on the
+  cluster-level services from which they are derived. A supercluster service
+  will be exposed with a regular supercluster IP if any source cluster-level
+  service specifies a cluster IP. A supercluster service will be headless if and
+  only if all cluster-level services from which it is derived are headless.
+- `NodePort` and `LoadBalancer`: These create `ClusterIP` services that would
+  sync as expected. For example If you export a `NodePort` service, the
+  resulting cross-cluster service will still be a supercluster IP type. The
+  local service will not be affected. Node ports can still be used to access the
+  cluster-local service in the source cluster, and only the supercluster IP will
+  reoute to endpoints in remote cluster.
+- `ExternalName`: It doesn't make sense to export an `ExternalName` service.
+  They can't be merged with other exports, and it seems like it would only
+  complicate deployments by even attempting to stretch them across clusters.
+  Instead, regular `ExternalName` type `Services` should be created in each
+  cluster individually. If a `ServiceExport` is created for an `ExternalName`
+  service, an `InvalidServiceType` condition will be set on the export.
+
+#### SuperclusterIP
+
+A non-headless `ServiceImport` is expected to have an associated IP address, the
+supercluster IP, which may be accessed from within an importing cluster. This IP
+may be supercluster-wide, or assigned on a per-cluster basis, but is expected to
+be consistent for the life of a `ServiceImport`. Requests to this IP from within
+a cluster will route to backends for the aggregated Service.
+
+Note: this doc does not discuss `NetworkPolicy`, which cannot currently be used
+to describe a policy that applies to a multi-cluster service.
+
+#### DNS
+
+_Optional, but recommended._
+
+When a `ServiceExport` is created, this will cause a domain name for the
+multi-cluster service to become accessible from within the supercluster. The
+domain name will be `<service>.<ns>.svc.supercluster.local`. MCS aims to align
+with the existing [service DNS
+spec](https://github.com/kubernetes/dns/blob/master/docs/specification.md). This
+section assumes familiarity with in-cluster Service DNS behavior.
+
+**SuperclusterIP services:** requests to this domain name from within an importing
+cluster will resolve to the supercluster IP, which points to endpoints for pods
+within the underlying `Service`(s) across the supercluster.
+
+**Headless services:** Within an importing cluster, the supercluster domain name
+will have multiple `A`/`AAAA` records, each containing the address of a ready
+endpoint of the headless service. `<service>.<ns>.svc.supercluster.local` will
+resolve to the set of all ready pod IPs for the service.
+
+Pods backing a supercluster service may be addressed individually using the
+`<hostname>.<clustername>.<svc>.<ns>.svc.supercluster.local` format. Necessary
```

Comment:
We need to think about total length here.

ns, svc, cluster, hostname can all be 63 characters, but a DNS FQDN can only be 253 TOTAL, including dots and suffix.

Reply:
Good point - added a note. Adding another label clearly increases the risk of hitting the FQDN limit. In a single cluster, we don't specifically limit the cluster domain do we? Is there any existing messaging around caution there?

Reply:
No, we have run fast-and-loose on this, on the assumption that the worst case of `<host>.<svc>.<ns>.svc.<suffix>` was unlikely to overflow.  This just puts us closer to the edge.  Maybe worth a bigger discussion with DNS folks about how to fail as gracefully as possible